### PR TITLE
Update gh actions

### PR DESCRIPTION
**Pull request by pckroon**
The GH workflow started complaining about soon-to-be deprecated node.js versions, so I figured I'd update the actions we use. In addition, I took the liberty to update the python versions we test against. In particular, I dropped py3.6 and added py3.10

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -124,7 +128,7 @@ jobs:
         --outdir dist/
         
     - name: Publish package to PyPI
-      uses: pypa/gh-action-pypi-publish@master      
+      uses: pypa/gh-action-pypi-publish@release/v1
```
Comment (fgrunewald) [inline .github/workflows/deploy.yml:131]: why this change? Does it work?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -59,11 +60,12 @@ jobs:
...
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.7
-      uses: actions/setup-python@v2
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
       with:
-        python-version: 3.9
+        python-version: 3.10
+        cache: pip
```
Comment (fgrunewald) [inline .github/workflows/deploy.yml:68]: what's the reason behind caching pip?

File: .github/workflows/run_tests.yml [hunk 3]
```diff
@@ -21,27 +21,30 @@ jobs:
     
     strategy:
       matrix:
-       py_version: ["3.6", "3.7", "3.8", "3.9", "3.6-dev", "3.7-dev", "3.8-dev", "3.9-dev"] 
+       py_version: ["3.7", "3.8", "3.9", "3.10", "3.7-dev", "3.8-dev", "3.9-dev", "3.10-dev"]
```
Comment (fgrunewald) [inline .github/workflows/run_tests.yml:24]: Here we can actually remove all the dev versions, because GHA will only collect the latest release version not the actual deve versions. This is an actions bug we discussed earlier I believe.

**Review by fgrunewald — CHANGES_REQUESTED** [review]
Looks good. Only thing is we should remove the devs because they are not actually running the devs. See [here](https://github.com/marrink-lab/polyply_1.0/issues/160). Could have been fixed with a new installer, but we should check.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (pckroon) [reply .github/workflows/deploy.yml:147]: Because I don't want to run off of their master branch. I'd rather depend on specific, released versions.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (pckroon) [reply .github/workflows/deploy.yml:72]: Because I think that should speed things up a little. There is a catch though, I'm not 100% sure we'll ever get the newest versions of our dependencies this way, since we don't list explicit version numbers in our requirements. And those requirements files are used to create a hash-index for the cache. Maybe rip it out again after all?

**Review by fgrunewald — APPROVED** [review]

**Comment by fgrunewald** [comment]
@pckroon I think this one is good to go

**Comment by pckroon** [comment]
Feel free to merge when the CI finishes.

**Review by fgrunewald — APPROVED** [review]
